webapp/packages/plugin-administration/src/ConfigurationWizard/ServerConfiguration/ServerConfigurationFormPart.ts

```diff
@@ -0,0 +1,144 @@
+  constructor(
+    formState: IFormState<null>,
+    private readonly administrationScreenService: AdministrationScreenService,
+    private readonly serverConfigResource: ServerConfigResource,
+    private readonly productInfoResource: ProductInfoResource,
+    private readonly defaultNavigatorSettingsResource: DefaultNavigatorSettingsResource,
+    private readonly authProvidersResource: AuthProvidersResource,
+    private readonly passwordPolicyResource: PasswordPolicyResource,
+    private readonly passwordPolicyService: PasswordPolicyService,
+  ) {
+    super(formState, DEFAULT_STATE_GETTER());
+  }
+
+  isOutdated(): boolean {
+    return super.isOutdated() || this.serverConfigResource.isOutdated() || this.defaultNavigatorSettingsResource.isOutdated();
+  }
+
+  isLoaded(): boolean {
+    return super.isLoaded() && this.serverConfigResource.isLoaded() && this.defaultNavigatorSettingsResource.isLoaded();
+  }
+
+  protected async validate(
+    data: IFormState<IServerConfigurationFormPartState>,
+    contexts: IExecutionContextProvider<IFormState<IServerConfigurationFormPartState>>,
+  ) {
+    if (this.administrationScreenService.isConfigurationMode) {
+      await this.authProvidersResource.load(CachedMapAllKey);
+
+      if (this.authProvidersResource.has(AUTH_PROVIDER_LOCAL_ID)) {
+        await this.passwordPolicyResource.load();
+
+        const isNameValid = this.state.serverConfig.adminName && this.state.serverConfig.adminName.length >= ADMIN_USERNAME_MIN_LENGTH;
+        const isPasswordValid = this.passwordPolicyService.validatePassword(this.state.serverConfig.adminPassword ?? '');
+
+        if (!isNameValid || !isPasswordValid.isValid) {
+          ExecutorInterrupter.interrupt(contexts);
+        }
+      }
+    }
+  }
+
+  protected format() {
+    if (this.state.serverConfig.adminName) {
+      this.state.serverConfig.adminName = this.state.serverConfig.adminName.trim();
+    }
+
+    if (this.state.serverConfig.adminPassword) {
+      this.state.serverConfig.adminPassword = this.state.serverConfig.adminPassword.trim();
+    }
+
+    if (this.state.serverConfig.serverName) {
+      this.state.serverConfig.serverName = this.state.serverConfig.serverName.trim();
+    }
+
+    if (this.state.serverConfig.serverURL) {
+      this.state.serverConfig.serverURL = this.state.serverConfig.serverURL.trim();
+    }
+  }
+
+  protected override async saveChanges() {
+    if (!isObjectsEqual(this.state.navigatorConfig, this.initialState.navigatorConfig)) {
+      await this.defaultNavigatorSettingsResource.save(this.state.navigatorConfig);
+    }
+
+    await this.serverConfigResource.save(this.state.serverConfig);
+  }
+
+  protected override async loader() {
+    const config = await this.serverConfigResource.load();
+    const productInfo = await this.productInfoResource.load();
+    const defaultNavigatorSettings = await this.defaultNavigatorSettingsResource.load();
```

Comment:
Lets load data in parallel mode so user have a better UX

```suggestion
    [config, productInfo, defaultNavigatorSettings] = await Promise.all([
        this.serverConfigResource.load(),
        this.productInfoResource.load(),
        this.defaultNavigatorSettingsResource.load()
    ]);
```